test/tap/tests/unit/admin_disk_upgrade_unit-t.cpp

```
@@ -0,0 +1,581 @@
+ * an in-memory SQLite3DB.
+ */
+class TestDiskUpgrade {
+public:
+	ProxySQL_Admin *admin;
+
+	TestDiskUpgrade() {
+		void *mem = calloc(1, sizeof(ProxySQL_Admin));
+		admin = reinterpret_cast<ProxySQL_Admin*>(mem);
+		SQLite3DB *db_ = new SQLite3DB();
+		db_->open((char *)":memory:", SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_FULLMUTEX);
+		admin->configdb = db_;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)